orttraining/orttraining/core/graph/optimizer_builder.cc

```diff
@@ -16,5 +16,16 @@ void OptimizerBuilderRegistry::RegisterBuilders() {
   GetInstance().Register<SGDOptimizerBuilder>("SGDOptimizer");
 }
 
+Status IsMatchingTypeAndShape(
+    const onnxruntime::Tensor& tensor,
+    const int32_t element_type,
+    const std::vector<int64_t>& expected_shape) {
+  ORT_ENFORCE(tensor.GetElementType() == element_type);
```

Comment:
ORT_ENFORCE() throws, so the only status being returned by this function is Status::OK().
would it make more sense to return bool?

Reply:
changing ORT_ENFORCE to ORT_RETURN_IF_NOT to correctly return status